build-system/check-package-manager.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const getStdout = require('./exec').getStdout;
+const setupInstructionsUrl = 'https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-quick.md#one-time-setup';
+
+// Color formatting may not yet be available via gulp-util.
+function red(text) {return '\x1b[31m' + text + '\x1b[0m';}
+function cyan(text) {return '\x1b[36m' + text + '\x1b[0m';}
+function green(text) {return '\x1b[32m' + text + '\x1b[0m';}
+function yellow(text) {return '\x1b[33m' + text + '\x1b[0m';}
+
+
+/**
+ * @fileoverview Makes sure that packages are being installed via yarn
+ */
+function main() {
+  // Yarn is already used by default on Travis, so there is nothing more to do.
+  if (process.env.TRAVIS) {
+    return 0;
+  }
+
+  // If npm is being run, print a message and cause 'npm install' to fail.
+  if (process.env.npm_execpath.indexOf('yarn') === -1) {
+    console/*OK*/.log(red(
```

Comment:
Curious, what does the /*OK*/ comment do?

Reply:
We have a presubmit check that forbids the use of `console.log` in the AMP runtime. The `/*OK*/` bypasses that check, since we know that the build tools are only going to be run in the presence of a console.